### PR TITLE
Add test for check SSE reactive empty response data

### PR DESCRIPTION
**Pull request by jedla97**
### Summary

Adding the test based on scenario from https://github.com/quarkusio/quarkus/issues/35966. When developing it there are some issues so for now it's disabled (https://github.com/quarkusio/quarkus/issues/37033 and https://github.com/quarkusio/quarkus/issues/36986). 

Test should run on ocp as well but as it it disable I don't see much point to run it now.

Maybe it would be nice to add this scenario (with empty request) to http-advanced (non-reactive)? There is recent test for SSE but with this there will need be some small adjustment to it. WDYT?

Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Backport
- [x] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [x] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by michalvavrik** [comment]
> Maybe it would be nice to add this scenario (with empty request) to http-advanced (non-reactive)? There is recent test for SSE but with this there will need be some small adjustment to it. WDYT?

My impression is that RESTEasy Classic and SSE don't go together too well: https://github.com/quarkus-qe/quarkus-test-suite/pull/1511#issuecomment-1805335024

**Review by michalvavrik — CHANGES_REQUESTED** [review]
You checked off `follow the best practices https://cbea.ms/git-commit/` but in https://cbea.ms/git-commit/ I can see rule number 4. _Do not end the subject line with a period_ Can you please try to provide more descriptive PR title? I know it's hard, I hardly ever do it right, but I think you can do better. Thanks

**Comment by michalvavrik** [comment]
@jedla97 lgtm (+ please address note above on PR title), but let's wait to https://github.com/quarkusio/quarkus/pull/37035 and enable native test before this go in

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/SseEventUpdateResource.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    @Path("client-update")
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response clientUpdate() {
+        String host = ConfigProvider.getConfig().getValue("quarkus.http.host", String.class);
+        int port = ConfigProvider.getConfig().getValue("quarkus.http.port", Integer.class);
+        StringBuilder eventCapture = new StringBuilder();
+
+        WebTarget target = ClientBuilder.newClient().target("http://" + host + ":" + port + "/api/sse/server-update");
+        SseEventSource eventSource = SseEventSource.target(target).build();
+        eventSource.register(ev -> {
```
[the remark this answers is not in the document]
Comment (jedla97) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/SseEventUpdateResource.java]: Updated. Used `CopyOnWriteArrayList`. As the string is inside of lambda it's not possible to use string outside of it (unless it's use atomic). I look inside of Quarkus test coverage and it have only kotlin test which using `Collections.synchronizedList`.

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/SseEventUpdateResource.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import jakarta.ws.rs.sse.Sse;
+import jakarta.ws.rs.sse.SseEventSink;
+import jakarta.ws.rs.sse.SseEventSource;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+
+@Path("sse")
+public class SseEventUpdateResource {
+    public static final String DATA_VALUE = "random data value";
+    @Context
```
[the remark this answers is not in the document]
Comment (jedla97) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/SseEventUpdateResource.java:28]: Fixed

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/SseEventUpdateResource.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        eventSource.register(ev -> {
+            eventCapture.append("event: name=").append(ev.getName()).append(" data={").append(ev.readData())
+                    .append("} and is empty: ").append(ev.isEmpty()).append("\n");
+        }, thr -> {
+            eventCapture.append("Error: ").append(thr.getMessage()).append("\n")
+                    .append(Arrays.toString(thr.getStackTrace()));
+        });
+
+        eventSource.open();
+        LockSupport.parkNanos(5_000_000_000L);
```
[the remark this answers is not in the document]
Comment (jedla97) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/SseEventUpdateResource.java]: Updated. Look at the waiting more and there is some time needed to consume and work with events. But 1 second should be enough.

File: http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/SseEventUpdateResource.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        eventSource.close();
```
[the remark this answers is not in the document]
Comment (jedla97) [reply http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/SseEventUpdateResource.java]: I add try-with-resources. Personally tried when there is error status is not 200. It's caught in `eventSource.register` state. So believe it`s  inside logic which catch it. The eventsource operation should only throw `
IllegalArgumentException` when the some parameter is null. In this case it should not happen.

**Comment by jedla97** [comment]
@michalvavrik Title updated, Last push was just change commit typo. No problem waiting. Hope it will be quick

**Comment by michalvavrik** [comment]
@jedla97 I'm starting to think upstream PR won't go in anytime soon. Please rebase so that we can get this in.

**Comment by jedla97** [comment]
run tests

**Review by michalvavrik — APPROVED** [review]
